### PR TITLE
Added accessibility support.

### PR DESCRIPTION
**Pull request by dumptruckman**
These changes will help with screen reader support.

### CODE REVIEW

**Review by jossmac — APPROVED** [review]
Nice one @dumptruckman!

@chenglou would you mind merging?

**Comment by chenglou** [comment]
Thanks!

File: index.jsx [hunk 1]
```diff
@@ -15,6 +15,8 @@ export class Radio extends React.Component {
     return (
       <input
         {...this.props}
+        role="radio"
```
Comment (yuheiy) [inline index.jsx:18]: This role is [implicit ARIA semantics for `input[type=radio]`](https://www.w3.org/TR/html-aria/#input-radio), so should not be used.
And we can use `menuitemradio` as a role for the element. But if you do, we can not overwrite the role.

File: index.jsx [hunk 2]
```diff
@@ -15,6 +15,8 @@ export class Radio extends React.Component {
...
+        aria-checked={optional.checked}
```
Comment (yuheiy) [inline index.jsx:19]: And this attribute is also implicit ARIA semantics, if the `checked` attribute is used.

File: index.jsx [hunk 3]
```diff
@@ -38,7 +40,7 @@ export class RadioGroup extends React.Component {
 
   render() {
     const {Component, name, selectedValue, onChange, children, ...rest} = this.props;
-    return <Component {...rest}>{children}</Component>;
+    return <Component role="radiogroup" {...rest}>{children}</Component>;
```
Comment (yuheiy) [inline index.jsx:43]: In all cases we do not want to set this role for `RadioGroup`. In some cases `Component` prop may not be able to set its role (eg when it is `<p>` element). Or this role may be set for `RadioGroup`'s child element. Then we have to take the trouble to remove this. Like this: `<RadioGroup role={undefined} ...>`

**Review by yuheiy — COMMENTED** [review]
This is not good for accessibility. We should revert this commit. @dumptruckman @chenglou

**Pull request merged**